src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/ImmutablePropertyCollection.cs

```diff
@@ -19,8 +19,15 @@ internal abstract class ImmutablePropertyCollection<T> : IEnumerable<T> where T
 
         protected ImmutablePropertyCollection(IEnumerable<T> items)
         {
+            // While the majority of elements (items, properties, metadata) are guaranteed to be unique,
+            // Target Frameworks are not, filter out duplicates - NuGet uses the int-based indexer anyway.
+#pragma warning disable CA2214 // Do not call overridable methods in constructors
+            string keySelector(T i) => GetKeyForItem(i);
```

Comment:
Nit: By defining this as a local function, a new `Func<T, string>` will be created each time its used. If you define it as a `Func<T, string>` you'll just get one.

Reply:
Yeah that's pretty damn annoying and not obvious. The analzyer recommends the change.

Reply:
I wouldn't normally have bothered commenting, as I'm hoping a future compiler will "fix" this, but I thought since you were lazily creating the HashSet that you were allocation-resistant :)

Reply:
Yep this allocation will add up.

Reply:
I think we can reduce allocation here quite a bit more too. I'll file a PR.

Reply:
Filed #5257.